src/twisted/mail/test/test_options.py

```diff
@@ -45,43 +45,62 @@ def testAliases(self):
             ]
         )
 
-    def _endpointTest(self, service):
-        """
-        Use L{Options} to parse a single service configuration parameter and
-        verify that an endpoint of the correct type is added to the list for
-        that service.
-        """
-        options = Options()
-        options.parseOptions(["--" + service, "tcp:1234"])
-        self.assertEqual(len(options[service]), 1)
-        self.assertIsInstance(options[service][0], endpoints.TCP4ServerEndpoint)
-
     def test_endpointSMTP(self):
         """
         When I{--smtp} is given a TCP endpoint description as an argument, a
         TCPServerEndpoint is added to the list of SMTP endpoints.
         """
-        self._endpointTest("smtp")
+        options = Options()
+        options.parseOptions(["--smtp", "tcp:1234", "--no-pop3"])
+        service = makeService(options)
+        service.privilegedStartService()
+        service.startService()
```

Comment:
Your problem here is that you're hard-coding a port number of 1234 and then *actually* starting a real mail server to listen on a live socket.

You might need to add a reactor parameter for testing to `makeService` or something, and somehow thread through an argument so that it's not doing any real I/O, with `MemoryReactor`.